portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -29,20 +29,16 @@
 /** Servlet that returns some example content. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  static final String COMMENT_INPUT_ID = "user-comment";
-  static final String USERNAME_ID = "user-name";
-
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Get the comment text, timestamp, and user name
     String comment = request.getParameter("text");
-    if (comment == null || comment.equals("")) {
-      response.sendRedirect("/index.html");
-      return;
-    }
     long timestamp = System.currentTimeMillis();
-    String username = request.getParameter("name");
-    if (username == null || username.equals("")) {
+    String email = request.getParameter("email");
+    String username = request.getParameter("username");
+    if (comment == null || comment.equals("") || email == null || email.equals("")
+        || username == null || username.equals("")) {
+      // One of the necessary parameters was not set: redirect
       response.sendRedirect("/index.html");
```

Comment:
As mentioned earlier, we should return error codes. This is an invalid request, so should be HTTP code 400.